docs/spec_draft.md

```diff
@@ -1444,6 +1445,144 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.select_and_scatter
+
+### Semantics
+
+Runs a windowed selection `select` function over `operand` with shape
+`window_dimensions` and stride `window_strides` to produce an amount
+of selected locations whose shape matches `source`. The values in `source`
+are then scattered to the selected locations in output. Multiple scattered
+elements which land in the same output location are combined using the
+`scatter` function.
+
+The following diagram shows how elements in `results` are computed from
+`operand` and `source` using a concrete example.
+
+<p align="center">
+  <img src="spec_draft/select_and_scatter.svg" />
+</p>
+
+Refer to [stablehlo.reduce_window](#stablehloreduce_window) to know more about
+the semantics of `padding`.
+
+The order of reductions is implementation-defined, which means that `scatter`
+and `init_value` must form a monoid to guarantee that the operation produces
+the same results for all inputs on all implementations.
+
+More formally,
+ * Apply [stablehlo.reduce_window](#stablehloreduce_window) with the following
+   parameters to get `selected_values` as output.
+   * `inputs` $=$ { `operand` },
+   * `init_values` $=$ { identity value for `select` },
```

Comment:
I can see the problem with the formulation and this is not what  I intend to do.

Given,

1. In the [hlo_evaluator](https://github.com/tensorflow/tensorflow/blob/73d23daf5cce6a06653b9c65992b4a4717644765/tensorflow/compiler/xla/hlo/evaluator/hlo_evaluator_typed_visitor.h#L1778), the padding values are skipped and for the very first selection iteration a value is compared with itself.  

2. Per the the HLO semantics,  the `init_value` parameter is to be used as the initial value for the scatter operation. , not the select function.

I am tempted to skip the padding values. IMO, `reduceWindowOp` cannot be used as is, but we might have to borrow parts of reduceWindow's formulation to do scatter of the non-padded values.

Reply:
It'd be good to avoid duplicating these nontrivial formalisms, and towards that end we could handwave this similarly to `all_reduce`:

```
 * `selected_values = reduce_window_without_init(...)` with the following inputs:
   * `inputs` $=$ [ `operand` ].
   * `window_dimensions`, `window_strides`, and `padding` which are used as is.
   * `base_dilations` $=$ `windows_dilations` $=$ `[1, ..., 1]`.
   * `body` defined as:
     ```C++
     (tensor<E> arg0, tensor<E> arg1) -> tensor<E> {
      return select(arg0, arg1) ? arg0 : arg1;
     }
     ```
     where `E = element_type(operand)`.
  * where `reduce_window_without_init` works exactly like `reduce_window`, 
    except that the `schedule` of the underlying `reduce` doesn't include 
    init values.
```